src/compiler/scala/tools/nsc/tasty/bridge/TypeOps.scala

```diff
@@ -0,0 +1,464 @@
+        case _ =>
+          mkRefinedTypeWith(parent :: Nil, refinedCls, u.newScopeWith(decl))
+      }
+    }
+
+    def NormalisedBounds(tpe: Type, sym: Symbol)(implicit ctx: Context): Type = tpe match {
+      case bounds @ UnmergablePolyBounds() =>
+        unsupportedError(s"diverging higher kinded bounds: $sym$bounds")
+      case tpe: u.TypeBounds => normaliseBounds(tpe)
+      case tpe               => tpe
+    }
+
+    def AppliedType(tycon: Type, args: List[Type])(implicit ctx: Context): Type = {
+
+      def typeRefUncurried(tycon: Type, args: List[Type]): Type = tycon match {
+        case tycon: u.TypeRef if tycon.typeArgs.nonEmpty =>
+          unsupportedError(s"curried type application $tycon[${args.mkString(",")}]")
+        case _ =>
+          u.appliedType(tycon, args)
+      }
+
+      if (args.exists(tpe => tpe.isInstanceOf[u.TypeBounds] | tpe.isInstanceOf[LambdaPolyType])) {
+        val syms = mutable.ListBuffer.empty[Symbol]
+        def bindWildcards(tpe: Type) = tpe match {
+          case tpe: u.TypeBounds   => ctx.newWildcardSym(tpe).tap(syms += _).pipe(_.ref)
+          case tpe: LambdaPolyType => tpe.toNested
+          case tpe                 => tpe
+        }
+        val args1 = args.map(bindWildcards)
+        if (syms.isEmpty) typeRefUncurried(tycon, args1)
+        else u.ExistentialType(syms.toList, typeRefUncurried(tycon, args1))
+      }
+      else {
+        typeRefUncurried(tycon, args)
+      }
+
+    }
+  }
+
+  private[bridge] def mkRefinedTypeWith(parents: List[Type], clazz: Symbol, decls: u.Scope): Type =
+    u.RefinedType.apply(parents, decls, clazz).tap(clazz.info = _)
+
+  private def normaliseIfBounds(tpe: Type): Type = tpe match {
+    case tpe: u.TypeBounds => normaliseBounds(tpe)
+    case tpe               => tpe
+  }
+
+  private def normaliseBounds(bounds: u.TypeBounds): Type = {
+    val u.TypeBounds(lo, hi) = bounds
+    if (lo.isHigherKinded && hi.isHigherKinded) {
+      if (mergeableParams(lo, hi)) {
+        val nuLo = lo.resultType.upperBound.subst(lo.typeParams, hi.typeParams.map(_.ref))
+        lo.typeParams.foreach { sym =>
+          sym.owner.rawInfo.decls.unlink(sym)
+          sym.owner.rawInfo.members.unlink(sym)
+          sym.owner = noSymbol
+        }
+        u.PolyType(hi.typeParams, u.TypeBounds(nuLo, hi.resultType.upperBound))
+      }
+      else bounds match {
+        case u.TypeBounds(lo: LambdaPolyType, hi: LambdaPolyType) => u.TypeBounds(lo.toNested,hi.toNested)
+        case _                                                    => bounds
+      }
+    }
+    else if (hi.isHigherKinded)
+      u.PolyType(hi.typeParams, u.TypeBounds(lo.upperBound, hi.resultType.upperBound))
+    else if (lo.isHigherKinded)
+      u.PolyType(lo.typeParams, u.TypeBounds(lo.resultType.upperBound, hi.upperBound))
+    else
+      bounds
+  }
+
+  private[bridge] def resolveErasedTypeRef(ref: ErasedTypeRef)(implicit ctx: Context): Type = {
```

Comment:
this is not correct for converting an inner class of a class,  in another branch I instead convert Type => ErasedTypeRef